task/build-helm-chart-oci-ta/0.2/IMAGE_SUBSTITUTION_GUIDE.md

```diff
@@ -0,0 +1,126 @@
+# Image Substitution Guide for Build Helm Chart OCI-TA Task
+
+## Overview
+
+The `build-helm-chart-oci-ta` task supports **image substitution** - the ability to replace
+placeholder images in Helm chart templates with actual built images before packaging. This
+is useful in CI/CD pipelines where you want to:
+
+1. Keep placeholder images in your chart templates (e.g., `localhost/myapp`)
+2. Build actual images in your pipeline
+3. Replace the placeholders with the built images before packaging the chart
+
+## How It Works
+
+### 1. Chart Template Structure
+
+Your Helm chart templates should use placeholder images:
+
+```yaml
+# templates/deployment.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: myapp
+spec:
+  template:
+    spec:
+      containers:
+      - name: app
+        image: localhost/myapp
+      - name: sidecar
+        image: localhost/sidecar
+```
+
+### 2. Image Mappings Configuration
+
+Define the substitution mappings as JSON. **The task automatically sorts mappings by source
+image length (longest first) to prevent partial matches**, so you can provide them in any
+order:
+
+```json
+[
+  {
+    "source": "localhost/myapp",
+    "target": "quay.io/myorg/myapp"
```

Comment:
I think it would be easier to understand if we add tags to our examples:

```suggestion
    "target": "quay.io/myorg/myapp:on-pr-{{revision}}"
```